v2.go

```diff
@@ -71,6 +75,107 @@ func handleAuthorizationSignInRequest(conn *websocket.Conn, action *Action) {
 	writeQueuedActions(client)
 	action.Type = authorizationSignInSuccess
 	client.writeJSON(action)
+
+	undeliveredMessages, err := findUndeliveredMessages(client.ID)
+	if err != nil {
+		log.Println("failed to find undelivered message ids")
+	}
+
+	for _, undeliveredMessage := range undeliveredMessages {
+		action := constructDeliverMessageAction(undeliveredMessage)
+		client.writeJSON(action)
+	}
+}
+
+func handleMessagingSendRequest(conn *websocket.Conn, action *Action) {
+	client, err := getSignedInClient(conn)
+	if err != nil {
+		client.writeFailure(messagingSendFailure, []string{err.Error()})
+		return
+	}
+
+	content, ok := action.Payload["content"].(string)
+	if !ok {
+		client.writeFailure(messagingSendFailure, []string{"content is missing"})
+		return
+	}
+
+	messageGroupID, ok := action.Payload["message_group_id"].(float64)
+	if !ok {
+		client.writeFailure(messagingSendFailure, []string{"message_group_id is missing"})
+		return
+	}
+
+	senderID, ok := action.Payload["sender_id"].(float64)
+	if !ok {
+		client.writeFailure(messagingSendFailure, []string{"sender_id is missing"})
+		return
+	}
+
+	if client.ID != int(senderID) {
+		client.writeFailure(messagingSendFailure, []string{"sender_id mismatch"})
+		return
+	}
+
+	if !isClientInMessageGroup(int(senderID), int(messageGroupID)) {
+		client.writeFailure(messagingSendFailure, []string{"sender doesn't belong to message group"})
+		return
+	}
+
+	message, err := createMessage(content, int(messageGroupID), int(senderID))
+	if err != nil {
+		client.writeFailure(messagingSendFailure, []string{err.Error()})
+		return
+	}
+
+	messageGroupMemberIDs, err := findAllMessageGroupMemberIDs(message.MessageGroupID)
+	if err != nil {
+		client.writeFailure(messagingSendFailure, []string{err.Error()})
+		return
+	}
+
+	var recipientIDs []int
```

Comment:
@cglotr Just wondering, I guess you do this to be explicit right?

Reply:
@imranariffin Didn't get your point. Are you referring to  `recipientIDs`?